### PR TITLE
Fix GRU mkldnn kernel fail on look_table_v2

### PR DESCRIPTION
**Pull request by lidanqing-intel**
### PR types
Bug fixes 

### PR changes
Others 

### Describe
fusion_gru_mkldnn kernel failed on lookup_table_v2 3 dims output. Flatten fusion_gru input dims.

### CODE REVIEW

**Comment by lidanqing-vv** [comment]
@grygielski @wojtuss  I changed according to our discusion, please review

File: paddle/fluid/operators/fused/fusion_gru_op.cc [hunk 1]
```diff
@@ -30,31 +30,40 @@ void FusionGRUOp::InferShape(framework::InferShapeContext* ctx) const {
...
   OP_INOUT_CHECK(ctx->HasOutput("XX"), "Output", "XX", "fusion_gru");
   OP_INOUT_CHECK(ctx->HasOutput("Hidden"), "Output", "Hidden", "fusion_gru");
-
   auto x_dims = ctx->GetInputDim("X");
-  PADDLE_ENFORCE_EQ(x_dims.size(), 2,
-                    platform::errors::InvalidArgument(
-                        "Input(X)'s rank must be 2, but received input dim "
-                        "size is:%d, input dim is:[%s]",
-                        x_dims.size(), x_dims));
+  auto x_mat_dims = (x_dims.size() > 2 && x_dims[1] == 1)
```
Comment (wojtuss) [inline paddle/fluid/operators/fused/fusion_gru_op.cc]: Please, update the condition: `(x_dims.size() == 3 && x_dims[1] == 1)`

File: paddle/fluid/operators/fused/fusion_gru_op.cc [hunk 2]
```diff
@@ -30,31 +30,40 @@ void FusionGRUOp::InferShape(framework::InferShapeContext* ctx) const {
...
+                        ? framework::flatten_to_2d(x_dims, 1)
+                        : x_dims;
 
   auto wx_dims = ctx->GetInputDim("WeightX");
-  PADDLE_ENFORCE_EQ(wx_dims.size(), 2,
+  auto wx_mat_dims = (wx_dims.size() > 2 && wx_dims[1] == 1)
```
Comment (wojtuss) [inline paddle/fluid/operators/fused/fusion_gru_op.cc]: Please, update the condition: `(wx_dims.size() == 3 && wx_dims[1] == 1)`

File: paddle/fluid/operators/fused/fusion_gru_op.cc (hunk 1 above)
Comment (lidanqing-vv) [reply paddle/fluid/operators/fused/fusion_gru_op.cc]: Ok

Comment (lidanqing-vv) [reply paddle/fluid/operators/fused/fusion_gru_op.cc]: Done

File: paddle/fluid/operators/fused/fusion_gru_op.cc (hunk 2 above)
Comment (lidanqing-vv) [reply paddle/fluid/operators/fused/fusion_gru_op.cc]: Done

**Review by wojtuss — APPROVED** [review]
LGTM

**Review by wojtuss — APPROVED** [review]
LGTM

**Review by grygielski — APPROVED** [review]
LGTM

**Pull request merged**